src/main/java/com/kosherjava/zmanim/hebrewcalendar/TefilaRules.java

```diff
@@ -170,34 +170,31 @@ public boolean isTachanunRecitedShacharis(JewishCalendar jewishCalendar) {
-				|| holidayIndex == JewishCalendar.TISHA_BEAV || jewishCalendar.isIsruChag()
-				|| jewishCalendar.isRoshChodesh()
-				|| (!tachanunRecitedShivasYemeiHamiluim &&
-						((!jewishCalendar.isJewishLeapYear() && month == JewishDate.ADAR)
-								|| (jewishCalendar.isJewishLeapYear() && month == JewishDate.ADAR_II)) && day > 22)
-				|| (!tachanunRecitedWeekOfPurim &&
-						((!jewishCalendar.isJewishLeapYear() && month == JewishDate.ADAR)
-								|| (jewishCalendar.isJewishLeapYear() && month == JewishDate.ADAR_II)) && day > 10 && day < 18)
-				|| (jewishCalendar.isUseModernHolidays()
-						&& (holidayIndex == JewishCalendar.YOM_HAATZMAUT || holidayIndex == JewishCalendar.YOM_YERUSHALAYIM))
-				|| (!tachanunRecitedWeekOfHod && month == JewishDate.IYAR && day > 13 && day < 21)) {
-			return false;
-		}
-		return true;
-	}
+        return jewishCalendar.getDayOfWeek() != Calendar.SATURDAY
+                && (tachanunRecitedSundays || jewishCalendar.getDayOfWeek() != Calendar.SUNDAY)
+                && (tachanunRecitedFridays || jewishCalendar.getDayOfWeek() != Calendar.FRIDAY)
+                && month != JewishDate.NISSAN
+                && (month != JewishDate.TISHREI || ((tachanunRecitedEndOfTishrei || day <= 8)
+                && (!tachanunRecitedEndOfTishrei || (day <= 8 || day >= 22))))
+                && (month != JewishDate.SIVAN || ((!tachanunRecitedWeekAfterShavuos || day >= 7)
+                && (tachanunRecitedWeekAfterShavuos || day >= (!jewishCalendar.getInIsrael()
+                && !tachanunRecited13SivanOutOfIsrael ? 14 : 13))))
+                && (!jewishCalendar.isYomTov() || (jewishCalendar.isTaanis()
+                && (tachanunRecitedPesachSheni || holidayIndex != JewishCalendar.PESACH_SHENI))) // Erev YT is included in isYomTov()
+                && (jewishCalendar.getInIsrael() || tachanunRecitedPesachSheni || tachanunRecited15IyarOutOfIsrael
+                || jewishCalendar.getJewishMonth() != JewishDate.IYAR || day != 15)
+                && holidayIndex != JewishCalendar.TISHA_BEAV && !jewishCalendar.isIsruChag()
+                && !jewishCalendar.isRoshChodesh()
+                && (tachanunRecitedShivasYemeiHamiluim ||
+                ((jewishCalendar.isJewishLeapYear() || month != JewishDate.ADAR)
+                        && (!jewishCalendar.isJewishLeapYear() || month != JewishDate.ADAR_II)) || day <= 22)
+                && (tachanunRecitedWeekOfPurim ||
+                ((jewishCalendar.isJewishLeapYear() || month != JewishDate.ADAR)
+                        && (!jewishCalendar.isJewishLeapYear() || month != JewishDate.ADAR_II)) || day <= 10 || day >= 18)
+                && (!jewishCalendar.isUseModernHolidays()
+                || (holidayIndex != JewishCalendar.YOM_HAATZMAUT && holidayIndex != JewishCalendar.YOM_YERUSHALAYIM))
+                && (tachanunRecitedWeekOfHod || month != JewishDate.IYAR || day <= 13 || day >= 21);
+    }

```

Comment:
This just removes the need for and if-else

Reply:
Although, now that I look at it, the code looks different, IntelliJ might have taken liberties and done some unwanted stuff here. I doubt the logic has changes, but let me know if you want to revert this.